### PR TITLE
Configure MicroBuild to sign NuGet.Jobs

### PR DESCRIPTION
**Pull request by scottbommarito**
https://github.com/NuGet/Engineering/issues/1728

- Added `sign.thirdparty.targets`, a target shared by all projects in the solution, that configures all the third-party assemblies in the project to be signed.
- Added `MicroBuild` dependencies to projects that didn't have them.
- Added `sign.scripts.targets` to projects that have PowerShell scripts so that it signs them.
- Removed `Validation.Callback.Vcs`, which is no longer needed and would have needed to be signed.

Note: This doesn't sign `NuGetCDNRedirect`, but because it's an App Service we are not required to sign it yet.

### CODE REVIEW

File: NuGet.Jobs.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 15
-VisualStudioVersion = 15.0.27121.1
+# Visual Studio Version 16
```
Comment (joelverhagen) [inline NuGet.Jobs.sln]: What if I am using dev15 😓 ?

File: src/CopyAzureContainer/CopyAzureContainer.csproj [hunk 2]
```diff
@@ -71,9 +71,27 @@
...
+  <PropertyGroup>
+    <SignPath>..\..\build</SignPath>
+    <SignPath Condition="'$(BUILD_SOURCESDIRECTORY)' != ''">$(BUILD_SOURCESDIRECTORY)\build</SignPath>
+    <SignPath Condition="'$(NuGetBuildPath)' != ''">$(NuGetBuildPath)</SignPath>
+    <SignType Condition="'$(SignType)' == ''">none</SignType>
+  </PropertyGroup>
+  <Import Project="..\..\build\sign.targets" Condition="Exists('..\..\build\sign.targets')" />
+  <Import Project="$(SignPath)\sign.microbuild.targets" Condition="Exists('$(SignPath)\sign.microbuild.targets')" />
+  <ItemGroup>
+    <PowerShellScriptsToSign Include="InstallAzCopy.ps1"/>
```
Comment (joelverhagen) [inline src/CopyAzureContainer/CopyAzureContainer.csproj]: 👍

File: NuGet.Jobs.sln (hunk 1 above)
Comment (scottbommarito) [reply NuGet.Jobs.sln]: Wasn't aware of exactly what this did...will revert.

File: src/CopyAzureContainer/Scripts/PreDeploy.ps1 [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# This file is a placeholder--it does nothing.
+# "sign.scripts.targets", which is imported by the project from ServerCommon, requires that "nssm.exe", "Functions.ps1", "PreDeploy.ps1", and "PostDeploy.ps1" all exist.
```
Comment (joelverhagen) [inline src/CopyAzureContainer/Scripts/PreDeploy.ps1]: Let's find a way to make this place-holder file unnecessary. Perhaps a no-op powershell or an ItemGroup `Remove="..."`

File: src/Validation.Common.Job/Validation.Common.Job.nuspec [hunk 4]
```diff
@@ -26,6 +26,7 @@
       <dependency id="NuGetGallery.Core" version="4.4.4-master-41290" />
       <dependency id="Serilog" version="2.5.0" />
       <dependency id="System.Net.Http" version="4.3.3" />
+      <dependency id="MicroBuild.Core" version="0.3.0" />
```
Comment (joelverhagen) [inline src/Validation.Common.Job/Validation.Common.Job.nuspec]: We should check IncludeAssets/PrivateAssets to exclude dev dependencies in the unit test requiring this change.

**Comment by joelverhagen** [comment]
This should go into `dev` not `master`.

**Comment by scottbommarito** [comment]
@joelverhagen - I thought the priority of this was enough to make it a hotfix-level change. I can change to `dev` if you think that's best however.

**Comment by joelverhagen** [comment]
@scottbommarito, that's a good point. We don't want to block this work and deployments on some issue in `dev`. Carry on 👍

**Pull request merged**